tests/e2e/tests/base.py

```diff
@@ -48,48 +47,66 @@ def get_expected_fx_pkg_str(self, os, alias, product_version):
+        :arg fx_pkg_name: The full Firefox package name that is the expected download.
+        :arg get_params: The GET params to pass to bouncer.
+        """
+        response = self.request_with_headers(base_url, params=get_params)
+        parsed_url = urlparse(response.url)
+        # verify service is up and a 200 OK is returned
+        assert requests.codes.ok == response.status_code, \
+            'Redirect failed with HTTP status. %s' % \
+            self.response_info_failure_message(base_url, get_params, response)
+        # verify download location
+        assert parsed_url.netloc in self.cdn_netloc_locations, \
+            'Failed, redirected to unknown host. %s' % \
+            self.response_info_failure_message(base_url, get_params, response)
+        # verify Firefox package name and version
+        assert fx_pkg_name in response.url, \
+            'Failed: Expected product str did not match what was returned %s' % \
+            self.response_info_failure_message(base_url, get_params, response)
+
+    def request_with_headers(self, url, params, user_agent=_user_agent_firefox, locale='en-US'):
+        """Make a request that includes 'user-agent', 'accept-language', and 'Connection: close' attributes in the
+        request header. Redirects will be followed.
+        :param url: URL of server to make the request against
+        :param params: GET parameters dict
+        :param user_agent: Browser useragent string
+        :param locale: header accept-language string
+        :return: response object
+        """
         headers = {'user-agent': user_agent,
                    'accept-language': locale,
                    'Connection': 'close'}
-
         try:
-            r = requests.head(url, headers=headers, verify=False, timeout=15,
-                              params=params, allow_redirects=False)
+            response = requests.head(url, headers=headers, verify=False, timeout=15, params=params, allow_redirects=False)
         except requests.RequestException as e:
-            request_url = self._build_request_url(url, params)
-
-            raise AssertionError('Failing URL: %s.\nError message: %s' % (request_url, e))
+            request_url = '%s/?%s' % (url, urlencode(params))
+            raise AssertionError('Failing URL: %s Error message: %s' % (request_url, e))
 
-        if r.status_code == 302 and r.headers['Location']:
+        if response.status_code == 302 and response.headers['Location']:
+            # TODO: I'm not sure why allow_redirects=False is being used. It seems we could set this to True
             try:
-                request_url = r.headers['Location']
-                r = requests.head(request_url, headers=headers, verify=False,
-                                  timeout=15, allow_redirects=True)
+                request_url = response.headers['Location']
+                response = requests.head(request_url, headers=headers, verify=False, timeout=15, allow_redirects=True)
```

Comment:
I was thinking the same. It's an interesting bit of code that I wasn't sure why we (possibly me) wrote. I'm leaning towards allow `allow_redirects=True`